app/Controllers/Graph.hs

```diff
@@ -48,14 +48,7 @@ getGraphJSON :: ServerPart Response
 getGraphJSON = do
     graphName <- lookText' "graphName"
     response <- liftIO $ getGraph graphName
-    withDefault response
-    where
-        withDefault (Just response) = return response
-        withDefault Nothing = return $
-            createJSONResponse $
-            object ["texts" .= ([] :: [Text]),
-                    "shapes" .= ([] :: [Text]),
-                    "paths" .= ([] :: [Text])]
+    return $ maybe (createJSONResponse $ object ["texts" .= ([] :: [Text]), "shapes" .= ([] :: [Text]), "paths" .= ([] :: [Text])]) createJSONResponse response
```

Comment:
This is better but can still be improved further.

1. `createJSONResposne` shouldn't be part of the function call. I realize here that `fromMaybe` is probably better to use than `maybe`.
2. You don't need to keep it all on one line. What I meant by my earlier comment is to use a single _expression_, but in general you can use line breaks to have that expression span multiple lines.

Reply:
Oh I see!
I used `fromMaybe` instead of `maybe` now. I am not sure that I interpreted your comment correctly, but I moved `createJSONResponse` outside of the `fromMaybe` call.